src/gerbil/boot/gx-init-exe.scm

```diff
@@ -17,10 +17,19 @@
 (define __gx#method-ref #f)
 
 (define (_gx#load-runtime!)
+  (define (gxi_GERBIL_HOME)
+    (let* ((result.string (shell-command "gxi --home" #t))
+           (result (car result.string))
+           (home (cdr result.string)))
```

Comment:
this might contain an extra newline.

Reply:
ah nm, it's `echo -n` in `gxi`.

Reply:
> ah nm, it's `echo -n` in `gxi`.

Did that on purpose. simplest thing that could possibly work.

> For the sake of Nix, you'll have to take the path to gxi from a suitable variable that may default to gxi but might be a full path. Actually, in Nix this would somehow return a fixed path without shelling out.

GIve me a Nix box and I'll give it a go. I purposely left it up to PATH for a number of reasons, and kept it very basic as it is just meant to work with .deb packages to start. Hard links, which I do all the time, really mess up this simple detection.

Reply:
The Nix build script is in the [nixpkgs](https://github.com/NixOS/nixpkgs) repo under `pkgs/development/compilers/gerbil/build.nix` and will somehow have to replace the definition of `gxi_GERBIL_HOME` to substitute to it the path from nix's `$out` parameter. Please make that easy to do somehow (and maybe even send me a patch for `build.nix`). Same fix needed in the `gxi` script.